### ISSUE TITLE
Karpenter is massively consolidate nodes on version 0.31.4

### ISSUE THREAD

**Issue by ariretiarno** [kubernetes-sigs/karpenter#1242]
### Description

**Observed Behavior**:
I have upgrade our karpenter controller since compability issue on kubernetes version. Previously  i'm using karpenter 0.28.1 and consolidate is work as expected. On karpenter 0.28.1, If there are new workloads, karpenter will spawn new nodes and doesn't move old workloads to these new node. But on this version, karpenter moving old workloads to new nodes, and the old nodes is cordoned and drained by karpenter. This behaviour causes Pod Evicted and Redeploy pods into new nodes which is on some apps is not acceptable.

Second behaviour, there are no new workloads, karpenter spawn new nodes and drained the old nodes and moving pods to new nodes. and if i check, node that lauched by karpenter is same type.

**Expected Behavior**:
Karpenter not massively spawn new nodes and move workloads into the new one. 

**Reproduction Steps** (Please include YAML):
provisioner.yaml
```
apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  name: t3a-large-common
spec:
  requirements:
    - key: karpenter.k8s.aws/instance-category
      operator: In
      values: [m, r]
    - key: karpenter.k8s.aws/instance-generation
      operator: Gt
      values: ["2"]
    - key: evermos.com/serviceClass
      operator: In
      values: ["t3a-large-common", "t3a-large-everpro", "t3a-large-data", "sonarqube", "t3a-large-etl", "t3a-medium-helm", "t3a-xlarge-observability"]
    - key: evermos.com/instanceType
      operator: In
      values: ["SPOT"]
    - key: evermos.com/zone
      operator: In
      values: ["ap-southeast-1a"]
    - key: "karpenter.sh/capacity-type" 
      operator: In
      values: ["spot"]
    - key: "karpenter.k8s.aws/instance-hypervisor"
      operator: In
      values: ["nitro"]
    - key: "karpenter.k8s.aws/instance-cpu"
      operator: In
      values: ["2", "4"]
    - key: "karpenter.k8s.aws/instance-size"
      operator: NotIn
      values: ["nano", "micro", "medium", "small"]
    - key: "node.kubernetes.io/instance-type"
      operator: NotIn
      values: ["m6i.large", "r6i.xlarge", "m5.large", "r6idn.xlarge", "r6in.xlarge", "r5.large", "r6a.large", "m7i-flex.large"]
    - key: karpenter.k8s.aws/instance-generation
      operator: NotIn
      values: ["7"]
    - key: "kubernetes.io/arch"
      operator: In
      values: ["amd64"]
  startupTaints:
  - key: node.cilium.io/agent-not-ready
    value: "true"
    effect: NoExecute
  providerRef:
    name: default
  kubeletConfiguration:
    maxPods: 40
  consolidation:
    enabled: true
```

**Versions**:
- Chart Version:v0.31.4
- Kubernetes Version (`kubectl version`): v1.26.15-eks-adc7111

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
Do you have logs from Karpenter that you can share from this behavior? It would also be good to see the AWSNodeTemplate that you are referencing here, and the pods that are getting scheduled and moved over when a new node is launched.

**Comment by ariretiarno** [comment]
Solved after upgrade to karpenter v0.32.9

Thankyou

**Comment by ariretiarno** [comment]
Solved after upgrade to karpenter v0.32.9
